src/atomsWithQuery.ts

```diff
@@ -0,0 +1,55 @@
+import {
+  ApolloClient,
+  OperationVariables,
+  QueryOptions,
+  ApolloQueryResult,
+} from '@apollo/client'
+import { Getter, WritableAtom } from 'jotai'
+import { clientAtom } from './clientAtom'
+import { createAtoms } from './common'
+
+type QueryArgs<
+  Variables extends object = OperationVariables,
+  Data = any
+> = QueryOptions<Variables, Data>
+
+type AtomWithQueryAction = {
+  type: 'refetch'
+}
+
+export const atomsWithQuery = <
+  Data,
+  Variables extends object = OperationVariables
+>(
+  getArgs: (get: Getter) => QueryArgs<Variables, Data>,
+  getClient: (get: Getter) => ApolloClient<any> = (get) => get(clientAtom)
+): readonly [
+  dataAtom: WritableAtom<Data, AtomWithQueryAction>,
+  statusAtom: WritableAtom<ApolloQueryResult<Data>, AtomWithQueryAction>
+] => {
+  return createAtoms(
+    getArgs,
+    getClient,
+    (client, args) => {
+      return {
+        subscribe: (observer) => {
+          client
+            .query(args)
+            .then((result) => {
+              observer.next?.(result)
+            })
```

Comment:
This means we can get the result only once, right?
Maybe it's how apollo works (it's the same before this PR.)

Reply:
Tried before, and as far as i remember this is the way it works, Apollo is not as nice as urql in API!